pkg/apis/management.cattle.io/v3/authz_types.go

```diff
@@ -106,26 +106,76 @@ type GlobalRoleBinding struct {
 }
 
 // +genclient
-// +kubebuilder:skipversion
 // +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// RoleTemplate holds configuration for a template that is used to create roles and clusterRoles for a cluster or project.
 type RoleTemplate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	DisplayName           string              `json:"displayName,omitempty" norman:"required"`
-	Description           string              `json:"description"`
-	Rules                 []rbacv1.PolicyRule `json:"rules,omitempty"`
-	Builtin               bool                `json:"builtin" norman:"nocreate,noupdate"`
-	External              bool                `json:"external"`
-	Hidden                bool                `json:"hidden"`
-	Locked                bool                `json:"locked,omitempty" norman:"type=boolean"`
-	ClusterCreatorDefault bool                `json:"clusterCreatorDefault,omitempty" norman:"required"`
-	ProjectCreatorDefault bool                `json:"projectCreatorDefault,omitempty" norman:"required"`
-	Context               string              `json:"context" norman:"type=string,options=project|cluster"`
-	RoleTemplateNames     []string            `json:"roleTemplateNames,omitempty" norman:"type=array[reference[roleTemplate]]"`
-	Administrative        bool                `json:"administrative,omitempty"`
+	// DisplayName is the human-readable name used for this resource.
```

Comment:
Nit: Might be good to indicate that this is what you see when you use the UI (i.e. add "displayed on the UI" to the end).

Reply:
I see UI as a consumer of the API, so it seems odd to put that information here. Since this would mean the code comments should be updated based on any changes in the UI's usage.

Reply:
updated